### PR TITLE
Add examples from FAFO

### PR DESCRIPTION
**Pull request by richardpringle**
- **Some docs for programs development (#1302)**
- **Add Balance Interface to Simulator (#1305)**
- **update amm test (#1310)**
- **Dev-container for find-out Friday**
- **Docs as github-pages**
- **Remove out-of-date READMEs**
- **Add minimal wasmlanche-sdk README**
- **Major wasmlanche-docs update**
- **Update `public` macro to not mention implementation details**
- **simulator docs (#1311)**
- **Don't ignore .so files, that's silly**
- **Fix Balance Method Ptrs (#1327)**

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// For format details, see https://aka.ms/devcontainer.json.
```
Comment (samliok) [inline .devcontainer/devcontainer.json:1]: should delete before merging

File: x/programs/rust/examples/find-out/src/lib.rs
Comment (samliok) [inline x/programs/rust/examples/find-out/src/lib.rs:1]: find-out should be renamed to counter, and the old counter crate program should be deleted

**Review by samliok — CHANGES_REQUESTED** [review]

File: x/programs/rust/examples/automated-market-maker/src/lib.rs [hunk 2]
```diff
@@ -19,16 +19,24 @@ const MAX_GAS: Gas = 10000000;
...
+    );
+
     context
         .store((
             (TokenX, token_x),
             (TokenY, token_y),
-            (LiquidityToken, liquidity_token),
+            (LiquidityToken, lt_program),
         ))
         .expect("failed to set state");
```
Comment (samliok) [inline x/programs/rust/examples/automated-market-maker/src/lib.rs:37]: this should be last line, probably a weird rebase

Comment (richardpringle) [reply x/programs/rust/examples/automated-market-maker/src/lib.rs:37]: done

**Review by samliok — APPROVED** [review]

**Pull request merged**